test/shared/src/main/scala/zio/test/package.scala

```diff
@@ -127,17 +129,39 @@ package object test extends CompileVariants {
       loop(fragment, FailureDetails(::(AssertionValue(assertion, value), Nil)))
     }
 
+  /**
+   * Checks the assertion holds for the given value.
+   */
+  @deprecated("use curried version", "1.0.0")
```

Comment:
```suggestion
  @deprecated("To benefit from much better type inference and type safety, we recommend that you use the curried version of assert, which takes two parameter lists instead of one: assert(value)(assertion)", "1.0.0")
```